### ISSUE TITLE
Some threads continue to run in background after Ctrl-C causing panic on re-run

### ISSUE THREAD

**Issue by gavofyork** [ethcore/parity#848]
Run parity, wait a while, ctrl-c, `ps aux` and you'll see it's still going.

**Comment by petevine** [comment]
I've noticed it can take a long time between Ctrl-C and the actual shutdown on ARM. (not sure if it's the slow write speeds or low processing power)

A message like:
`Closing, please wait`

could reassure the user everything's ok.

**Comment by peterjoel** [comment]
Is this what causes the intermittent panic:
```
thread '<main>' panicked at 'called `Result::unwrap()` on an `Err` value: 
Util(StdIo(Error { repr: Os { code: 48, message: "Address already in use" } }))', 
src/libcore/result.rs:746
```
?

**Comment by gavofyork** [comment]
> It shouldn't panic.

yes indeed. it should bomb out gracefully with a similar message.

**Comment by tomusdrw** [comment]
Most of the issues here are addressed by #893. JSON-RPC threads should also be fixed in #901.
Closing for now. Reopen or file another issue if problem occurs again.
